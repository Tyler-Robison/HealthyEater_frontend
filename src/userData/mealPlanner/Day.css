.Day {
    max-width: 500px !important;
    height: 100%;
    border: 1px solid black;
    position: relative;
    background-color: white;
    min-height: 40rem;
}

@media (max-width: 1400px) {
    .Day {
       min-height: 45rem;
     
    }
}

@media (max-width: 1000px) {
    .Day {
       min-height: 12rem;
     
    }
}

.Day-points {
    position: absolute;
    bottom: 0rem;
    margin: 0px;
    left: 50%;
    transform: translate(-50%, -50%);
}