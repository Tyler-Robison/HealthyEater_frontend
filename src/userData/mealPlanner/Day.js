import React, { useContext } from "react";
import GlobalContext from "../../context/GlobalContext";
import PlannerAPI from "../../APIs/plannerAPI";
import { Link } from "react-router-dom";
import './Day.css'
import { v4 as uuid } from 'uuid';


const Day = ({ day }) => {
    const { currentUser, setCurrentUser, token } = useContext(GlobalContext)

    const deleteMeal = async (meal_id) => {
        const res = await PlannerAPI.deleteMeal(currentUser.id, meal_id, token)

        currentUser.mealplan = currentUser.mealplan.filter(m => m.id !== res.id)

        setCurrentUser({ ...currentUser, currentUser })
    }


    // day = 'Mon', 'Tues' etc
    const daysMeals = currentUser.mealplan.filter(meal => day === meal.day)

    let totalPoints = 0;
    for (let ele of daysMeals) {
        totalPoints += ele.ww_points;
    }

    return (
        <div className='Day col-lg'>
            <p><b>{day}</b></p>

            <div className="list-container">
                <ol>
                    {daysMeals.map(d => {
                        return <li key={uuid()}> <Link to={`/${d.recipe_id}/detail`} className='RecipeList-para'><b>{d.name}</b></Link> ({d.ww_points ? d.ww_points : 'Not Available'} points)
                            <button data-testid={d.id} className="general-btn-red small-btn" onClick={() => deleteMeal(d.id)}>X</button>
                        </li>
                    })}
                </ol>
            </div>
            {totalPoints > 0 && <p className="Day-points"><b>Points:</b> {totalPoints}</p>}
        </div>
    )
}

export default Day;