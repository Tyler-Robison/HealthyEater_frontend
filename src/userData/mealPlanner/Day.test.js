import { render, screen } from '@testing-library/react';
import Day from './Day';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext';


it("renders without crashing", function () {
    <MemoryRouter>
        <Day day='Mon' />
    </MemoryRouter>
});


it("matches snapshot on day with meals", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <Day day='Mon' />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

it("matches snapshot on day with no meals", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <Day day='Wed' />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});