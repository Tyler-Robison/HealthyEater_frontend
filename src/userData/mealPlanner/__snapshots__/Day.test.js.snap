// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on day with meals 1`] = `
<DocumentFragment>
  <div
    class="Day col-lg"
  >
    <p>
      <b>
        Mon
      </b>
    </p>
    <ol>
      <li>
         
        <a
          class="RecipeList-para"
          href="/1111/detail"
        >
          <b>
            bacon
          </b>
        </a>
         (20 points)
        <button
          class="general-btn-red small-btn"
          data-testid="1"
        >
          X
        </button>
      </li>
      <li>
         
        <a
          class="RecipeList-para"
          href="/2222/detail"
        >
          <b>
            eggs
          </b>
        </a>
         (10 points)
        <button
          class="general-btn-red small-btn"
          data-testid="2"
        >
          X
        </button>
      </li>
    </ol>
    <p
      class="Day-points"
    >
      <b>
        Total Points:
      </b>
       30
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot on day with no meals 1`] = `
<DocumentFragment>
  <div
    class="Day col-lg"
  >
    <p>
      <b>
        Wed
      </b>
    </p>
    <ol />
  </div>
</DocumentFragment>
`;
